.github/workflows/report-on-vulnerabilities.yaml

```diff
@@ -16,15 +16,50 @@ jobs:
     runs-on: ubuntu-latest
     outputs:
       results: ${{ steps.parse-results.outputs.results }}
-    steps:
-    - name: Scan for vulnerabilities
+    steps:    
+    - name: Get Branches Name
+      id: get-branches
+      shell: bash
+      run: |
+        all_branches=$(curl -s https://api.github.com/repos/${{ env.IMAGE_NAME }}/branches | jq -r '.[].name | select(startswith("release-"))' | sort -rV | head -n 2)
+        releasebranch1=$(echo "$all_branches" | sed -n 1p)
+        releasebranch2=$(echo "$all_branches" | sed -n 2p)
+
+        echo "releasebranch1=$releasebranch1" >> $GITHUB_OUTPUT
+        echo "releasebranch2=$releasebranch2" >> $GITHUB_OUTPUT
```

Comment:
I see a potential problem with this - say later the release branch is cut, this job will run from main and release branch(es) every day which causes duplicate scans. Ideally we should target the current branch and run this job only for that branch.

Reply:
What I have understood is:
1. Current Problem: Duplicate Scans
2. To address this, we need to scan the latest version daily for vulnerabilities, similar to how it was done before this PR.
3. For the n-1 and n-2 versions, we should run the scan only once since no new changes are made after release.

Reply:
GitHub Actions only run from the default branch, which is `main` in our case, so this isn't a concern.

Reply:
@chipzoller @realshuting so, i don't have to change anything here?